My Jetpack: Adding product tour

grzegorz-cp opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes MARTECH-52

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* restoring product tour added during the previous project iteration
* connected to: https://github.com/Automattic/jetpack/pull/42770

<img width="422" alt="SCR-20250327-qplg" src="https://github.com/user-attachments/assets/00c5391b-5e46-419e-98a0-c0b5886dbe5a" />


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Spin the PR with new JN site
* go to My Jetpack
* add `from=jetpack-onboarding` parameter
* verify that the tour doesn't show up

grzegorz-cp committed:
My Jetpack: Restoring the tour

grzegorz-cp committed:
changelog

Initsogar reviewed (dismissed):
Looking great! 👍🏽

keoshi reviewed (dismissed):
Works and feels great. All the interactions are there too.

Minor cosmetic things:

- Titles should be using [Core’s `heading-x-large()`](https://wordpress.github.io/gutenberg/?path=/docs/tokens-typography--page)
- The buttons should also use [Core's buttons](https://wordpress.github.io/gutenberg/?path=/story/components-button--primary) with the appropriate padding.
- The SVG in the close button should be `24x24px`.
- Spacing in between title and body copy should be `8px`.

![image](https://github.com/user-attachments/assets/0ac510f7-248d-435b-a8d5-8857706ef83e)


- Padding for the action area should be `16px`.

![image](https://github.com/user-attachments/assets/9679a80c-a535-4d0e-bca5-964197e8785c)


- Dots should be much closer to each other and replicate the onboarding to the block editor.

Jetpack | Block editor
--|--
<img width="139" alt="image" src="https://github.com/user-attachments/assets/6f32a7b3-a272-4f6f-b01c-451f5dab0a69" /> | <img width="121" alt="image" src="https://github.com/user-attachments/assets/8e5e6f54-d24e-4886-9739-447d2e879b84" />

Finally, as @Initsogar suggested, we could increase the height a tiny bit so that the third one doesn't cause any layout shift.

Thank you!

grzegorz-cp commented:
Thanks @keoshi, I'll implement the changes.
This tour is using existing Gude component and doesn't allow replacing buttons but I will adjust CSS.

I reduced the size of the SVG because when it's 24px the icon is much larger than the designs. The spacing between the icon and the edge of SVG is smaller than the design. I'll keep it as it is but if you confirm 24px is better I can update is as well.

grzegorz-cp commented:
> Just so I understand, is this the component used as the onboarding in the block editor?
Yes, it's the same component with a bit of extra CSS. [link](https://wordpress.github.io/gutenberg/?path=/docs/components-guide--docs)

grzegorz-cp committed:
Merge branch 'trunk' into update/martech-52-restore-my-jetpack-tour

grzegorz-cp committed:
CSS updates

grzegorz-cp commented:
It's ready for review when you get back @keoshi

grzegorz-cp committed:
Avoiding layout shifts

grzegorz-cp committed:
Linting

grzegorz-cp committed:
Merge branch 'trunk' into update/martech-52-restore-my-jetpack-tour

grzegorz-cp committed:
Linting

grzegorz-cp committed:
Linting

grzegorz-cp committed:
Merge branch 'trunk' into update/martech-52-restore-my-jetpack-tour

grzegorz-cp commented:
I'm going to dismiss your review @keoshi because you are away and I believe I addressed all issues. Let's check in once you are back if anything else needs updates.
Since @Initsogar approved it previously, I'll ship the PR today.
